src/pkg/ocidriver/ocidriver.go

```diff
@@ -0,0 +1,293 @@
+package ocidriver
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/manifest/manifestlist"
+	"github.com/docker/distribution/manifest/schema2"
+	"github.com/docker/distribution/reference"
+	v2client "github.com/docker/distribution/registry/client"
+	commonhttp "github.com/goharbor/harbor/src/common/http"
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+)
+
+// OciRegistry ...
+type OciRegistry struct {
+	baseURL   string
+	transport http.RoundTripper
+}
+
+// OciRepository ...
+type OciRepository struct {
+	Registry *OciRegistry
+
+	Name string
+}
+
+// OciImage ...
+type OciImage struct {
+	Registry   *OciRegistry
+	repository *OciRepository
+
+	Name   string
+	Digest digest.Digest
+}
+
+// OciManifest ...
+type OciManifest struct {
+	*schema2.DeserializedManifest
+
+	repository *OciRepository
+}
+
+// OciManifestList ...
+type OciManifestList struct {
```

Comment:
should we define this as index instead of Manifestlist to align with OCI spec?

Reply:
change to OciManifestIndex